runtime-tools-quarkus-extension-parent/runtime/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/forms/impl/FormsStorageImpl.java

```diff
@@ -17,59 +17,163 @@
 package org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.impl;
 
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.FilenameUtils;
-import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.FormInfo;
+import org.apache.commons.io.IOUtils;
 import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.FormsStorage;
+import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.model.Form;
+import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.model.FormConfiguration;
+import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.model.FormFilter;
+import org.kie.kogito.runtime.tools.quarkus.extension.runtime.forms.model.FormInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.json.JsonObject;
 
 @ApplicationScoped
 public class FormsStorageImpl implements FormsStorage {
 
     private static final String FORMS_STORAGE_PATH = "/forms";
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(FormsStorageImpl.class);
+
     private URL formsFolder;
 
+    private Map<String, FormInfo> formInfoMap = new HashMap<>();
+
     @PostConstruct
     public void init() {
-        formsFolder = Thread.currentThread().getContextClassLoader().getResource(FORMS_STORAGE_PATH);
+        try {
+            URL originURL = Thread.currentThread().getContextClassLoader().getResource(FORMS_STORAGE_PATH);
+            URI originUri = originURL.toURI();
+            String newPath = originUri.getPath().replace("target/classes", "src/main/resources");
+            URI newURI = originUri.resolve(newPath);
+            formsFolder = newURI.toURL();
+            reloadFormBaseInfoList();
+        } catch (MalformedURLException e) {
+            LOGGER.warn(e.getMessage());
+        } catch (URISyntaxException e) {
+            LOGGER.warn(e.getMessage());
+        }
     }
 
     @Override
     public int getFormsCount() {
-        if (formsFolder == null) {
-            return 0;
+        return formInfoMap.keySet().size();
+    }
+
+    @Override
+    public Collection<FormInfo> getFormInfoList(FormFilter filter) {
+        final Collection<FormInfo> formInfoCollection = new ArrayList();
+        Collection<String> matchedNames = null;
+        if (filter != null && filter.getNames().size() > 0) {
```

Comment:
here you could do return formInfoMap.entrySet().filter().map().collect(); instead of having this intermediate matchedNames collection.